lib/archethic_web/controllers/api/web_hosting_controller/resources.ex

```diff
@@ -0,0 +1,175 @@
+defmodule ArchethicWeb.API.WebHostingController.Resources do
+  @moduledoc false
+
+  use ArchethicWeb, :controller
+
+  alias Archethic
+
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.TransactionData
+
+  use Pathex
+
+  require Logger
+  @addresses_key "addresses"
+  @metadata_key "metaData"
+  @aewebversion_key "aewebVersion"
+  @path_seperator "/"
+
+  def load(
+        txn = %Transaction{
+          address: last_address,
+          data: %TransactionData{content: content}
+        },
+        url_path,
+        cache_headers
+      ) do
+    with {:ok, json_content} <- Jason.decode(content),
+         {:ok, metadata, _aeweb_version} <- get_metadata(json_content),
+         {:ok, file, mime_type, resource_path} <- get_file(metadata, url_path),
+         {cached?, etag} <- get_cache(cache_headers, last_address, url_path),
+         {:ok, file_content, encoding} <- get_file_content(file, cached?, resource_path) do
+      {:ok, file_content, encoding, mime_type, cached?, etag}
+    else
+      {:error, %Jason.DecodeError{}} ->
+        {:error, :invalid_content}
+
+      {:error, :file_not_found} ->
+        {:error, :file_not_found}
+
+      {:error, :get_metadata} ->
+        {:error, "Error: Cant access metadata and aewebversion, RefTxn: #{last_address}"}
+
+      {:error, :is_a_directory} ->
+        {:error, {:is_a_directory, txn}}
+
+      error ->
+        error
+    end
+  end
+
+  def get_metadata(json_content) do
+    case json_content do
+      %{@metadata_key => metadata, @aewebversion_key => aewebversion} ->
+        {:ok, metadata, aewebversion}
+
+      _ ->
+        {:error, :get_metadata}
+    end
+  end
+
+  # index file
+  def get_file(metadata, []) do
+    case Map.get(metadata, "index.html", :error) do
+      :error ->
+        {:error, :is_a_directory}
+
+      value ->
+        {:ok, value, MIME.from_path("index.html"), "index.html"}
+    end
+  end
+
+  def get_file(metadata, url_path) do
+    resource_path = Enum.join(url_path, @path_seperator)
+
+    case Map.get(metadata, resource_path) do
+      nil ->
+        if is_a_directory(metadata, resource_path) do
+          {:error, :is_a_directory}
+        else
+          {:error, :file_not_found}
+        end
+
+      file ->
+        {:ok, file, MIME.from_path(resource_path), resource_path}
+    end
+  end
+
+  def get_file_content(_, true, _),
+    do: {:ok, nil, nil}
+
+  def get_file_content(
+        file_metadata = %{@addresses_key => address_list},
+        _cached? = false,
+        resource_path
+      ) do
+    resource_path = if resource_path == "", do: "index.html", else: resource_path
```

Comment:
I think the resource_path would never be an empty string as it is return by the function `get_file` which return either `Enum.join(url_path, @path_seperator)` or `"index.html"`